mllib/src/main/scala/org/apache/spark/ml/Predictor.scala

```diff
@@ -113,6 +113,7 @@ abstract class Predictor[
   private[ml] def featuresDataType: DataType = new VectorUDT
 
   override def transformSchema(schema: StructType): StructType = {
+    validateParams()
```

Comment:
In cases like this where the Estimator and Model share validateAndTransformSchema, the call could be moved to validateAndTransformSchema.